core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -229,9 +259,12 @@ private[spark] class Executor(
       // ClosedByInterruptException during execBackend.statusUpdate which causes
       // Executor to crash
       Thread.interrupted()
+      notifyAll()
```

Comment:
I was not looking at it from performance point of view, but from semantics.
notify implies we expect only single thread to be waiting, notifyAll implies we can expect multiple to be waiting.

If we are unsure if multiple threads could be waiting, then it is fine.